File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& v, int target) {
        int n = v.size();

        unordered_map<int,int> m;
        int compliment;

        for(int i=0 ; i<n ; i++){
        compliment = target - v[i];

        if(m.find(compliment) != m.end()){
            return {m[compliment],i};
        }

        m[v[i]] = i;
        }
        return {};
    }
};

File: 1-two-sum/two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        
        ans = {}
        for i in range(0,len(nums)):
            compliment = target - nums[i]
            if compliment in ans:
               return [ans[compliment],i]
            ans[nums[i]] = i
        

        return []